src/terms/nonlocal.jl

```diff
@@ -184,7 +184,14 @@ end
 Build form factors (Fourier transforms of projectors) for an atom centered at 0.
 """
 function build_form_factors(psp, qs)
-    qnorms = norm.(qs)
+
+    ## The AtomicNonlocal NaN boiled down to ForwardDiff + norm + StaticArrays at [0., 0., 0.]
+    ## https://github.com/JuliaDiff/ForwardDiff.jl/issues/243#issuecomment-369948031
+    ## specifically this happend because qs[1] is a Vec3 of zeros
+    ## TODO fix this more generally
+    # qnorms = norm.(qs)
+    qnorms = vcat(norm(Vector(qs[1])), norm.(@view qs[2:end]))
```

Comment:
Yikes ... actually there is no guarantee that the first qs is all-zero in theory (usually it is though).

Reply:
This is fixed by a new custom dispatch on `LinearAlgebra.norm(x::SVector{S,<:ForwardDiff.Dual})` in `forwarddiff_rules.jl`